DxDispatch/src/dxdispatch/DmlDispatchable.cpp

```diff
@@ -101,16 +142,255 @@ void FillBindingData(
     }
 }
 
+DmlDispatchable::BindPoints DmlDispatchable::GetBindPoints() const
+{
+    if (m_isSerializedGraph)
+    {
+        return m_bindPoints; 
+    }
+    else
+    {
+        const auto& desc = std::get<Model::DmlDispatchableDesc>(m_desc);
+        BindPoints result;
+        
+        for (const auto& input : desc.bindPoints.inputs)
+        {
+            result.inputs.push_back({input.name, input.resourceCount, input.required});
+        }
+        for (const auto& output : desc.bindPoints.outputs)
+        {
+            result.outputs.push_back({output.name, output.resourceCount, output.required});
+        }
+        
+        return result;
+    }
+}
+
+DmlDispatchable::BindPoints DmlDispatchable::GetSerializedBindPoints(const DmlSerializedGraphDesc& serializedDesc)
+{
+    BindPoints result;
+    
+    for (const auto& inputEdge : serializedDesc.InputEdges) 
+    {
+        result.inputs.push_back({inputEdge.Name, 1, true});
+    }
+    
+    for (const auto& node : serializedDesc.Nodes)
+    {
+        if (std::holds_alternative<DmlSerializedGraphNodeConstantVariant>(node.Desc))
+        {
+            const auto& constantVariant = std::get<DmlSerializedGraphNodeConstantVariant>(node.Desc);
+            if (std::holds_alternative<ConstantName>(constantVariant))
+            {
+                result.inputs.push_back({node.Name, 1, true});
+            }
+        }
+    }
+    
+    for (const auto& outputEdge : serializedDesc.OutputEdges)
+    {
+        result.outputs.push_back({outputEdge.Name, 1, true});
+    }
+    
+    return result;
+}
+
+std::unordered_map<std::string, DML_TENSOR_DATA_TYPE> DmlDispatchable::ExtractConstantDataTypes(const DmlSerializedGraphDesc& serializedDesc)
+{
+    std::unordered_map<std::string, DML_TENSOR_DATA_TYPE> constantDataTypes;
+    std::unordered_map<uint32_t, DmlIntermediateSerializedGraphEdge> constantNodeEdges;
+
+    for (const auto& edge : serializedDesc.IntermediateEdges) {
+        if (std::holds_alternative<DmlSerializedGraphNodeConstantVariant>(serializedDesc.Nodes[edge.FromNodeIndex].Desc)) {
```

Comment:
We should only consider ConstantName type edges because we will only binding large tensors during initialization.

Reply:
done